src/libs/ActiveClientManager/index.js

```diff
@@ -7,13 +7,28 @@ const clientID = Str.guid();
 const maxClients = 20;
 
 let activeClients;
+
+// Whether the current clientID is set
+let didInitialize = false;
+
+let resolveReadyPromise = null;
+
+// Keeps track of the ActiveClientManager's readiness.
+const readyPromise = new Promise((res) => {
+    resolveReadyPromise = res;
+});
+
 Onyx.connect({
     key: ONYXKEYS.ACTIVE_CLIENTS,
     callback: (val) => {
         activeClients = !val ? [] : val;
         if (activeClients.length >= maxClients) {
             activeClients.shift();
-            Onyx.set(ONYXKEYS.ACTIVE_CLIENTS, activeClients);
+            Onyx.set(ONYXKEYS.ACTIVE_CLIENTS, activeClients).then(() => {
+                if (didInitialize) {
+                    resolveReadyPromise();
```

Comment:
This is just an empty function, correct?

I'm struggling to determine the order of things...

Will it be:
 - the browser loads `index.js` which sets `resolveReadyPromise` to `NULL` and `didInitialize` to `false`
 - `Onyx.connect` gets called
   - on successful completion `Onyx.set` gets called
   - after `Onyx.set` is called, since `didInitialize` is `false` then we do not call `resolveReadyPromise`, so `isReady` is a non-resolved promise whenever it is called.
 - `init` gets called
   - `didInitialize` gets set to `true`
 - `isReady` gets called (from `Network.js`)
   - `readyPromise` is returned as an un-resolved promise
 - Client gets disconnected, and then re-connected
 - `Onyx.connect` gets called again
   - on successful completion `Onyx.set` gets called
   - after `Onyx.set` is called, since `didInitialize` is now set to `true` then we call `resolveReadyPromise` (which is an empty function) that resolves `readyPromise`
 - In `Network.js`, the `ActiveClientManager.isReady().then` can run because the `readyPromise` was resolved in `index.js`

Please correct me if any of that^ is wrong.

Reply:
> Client gets disconnected, and then re-connected. 
Apart from it, you are correct. It does not happen in the flow.

Reply:
If it does not happen when the client reconnects, when does `Onyx.connect` get called again?

Reply:
Don't bother by those they were related to old implementation. And things have changed since then. Now If you look at the top of the file, we have a subscription for NETWORK key which triggers Onyx.connect for NETWORK_REQUEST_QUEUE.

Reply:
But even step #8 in the PR description say "When the user comes back online"

I'm just wondering how/when `Onyx.connect` gets called again.

Reply:
> Now If you look at the top of the file, we have a subscription for NETWORK key which triggers Onyx.connect for NETWORK_REQUEST_QUEUE.

I have no idea what you're referring to...

Reply:
Ok...I believe the answer is that `Onyx.connect` gets called when `Onyx.merge` is called within the `init` function.

Reply:
Yes for activeCLientManager.